### PR TITLE
Add default `params_to_tune` for `DateFlagsTransform`, `TimeFlagsTransform`, `SpecialDaysTransform`, `FourierTransform`

### LINKED ISSUE #1221: Add default `params_to_tune` for timestamp transforms
### 🚀 Feature Request

Add default `params_to_tune` for timestamp transforms.

### Proposal

Suggested grid for `HolidayTransform`: None.

Suggested grid for `DateFlagsTransform`: 
```python
{
    "day_number_in_week": CategoricalDistribution([False, True]),
    "day_number_in_month": CategoricalDistribution([False, True]),
    "day_number_in_year": CategoricalDistribution([False, True]),
    "week_number_in_month": CategoricalDistribution([False, True]),
    "week_number_in_year": CategoricalDistribution([False, True]),
    "month_number_in_year": CategoricalDistribution([False, True]),
    "season_number": CategoricalDistribution([False, True]),
    "year_number": CategoricalDistribution([False, True]),
    "is_weekend": CategoricalDistribution([False, True]),
}
```

Suggested grid for `TimeFlagsTransform`: 
```python
{
    "minute_in_hour_number": CategoricalDistribution([False, True]),
    "fifteen_minutes_in_hour_number": CategoricalDistribution([False, True]),
    "hour_number": CategoricalDistribution([False, True]),
    "half_hour_number": CategoricalDistribution([False, True]),
    "half_day_number": CategoricalDistribution([False, True]),
    "one_third_day_number": CategoricalDistribution([False, True]),
}
```

Suggested grid for `SpecialDaysTransform`: 
```python
{
    "find_special_weekday": CategoricalDistribution([False, True]),
    "find_special_month_day": CategoricalDistribution([False, True]),
}
```

Suggested grid for `FourierTransform`: 
```python
{
    "order": IntLogUniform(low=1, high=max_value).
}
```
`max_value=ceil(period/2)`: this is the border value in our implementation.

To discuss:
- Do we want a grid for `HolidayTransform`?
- Are we ok, that all `False` are invalid set of parameters for `DateFlagsTransform`, `TimeFlagsTransform`, `SpecialDaysTransform`.

### Test cases

Tests analogous to tests for `params_to_tune` in models:
- Test that each parameter of the grid is correct:
  - Value can be sampled.
  - Value can be set inside the transform using `set_params`.
  - Calling `fit` works fine.

Check different `periods` values`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look at #1221.

## Closing issues
Closes #1221.

### CODE REVIEW

File: tests/test_transforms/test_timestamp/test_fourier_transform.py [hunk 1]
```diff
@@ -166,3 +167,17 @@ def test_forecast(ts_trend_seasonal):
 def test_save_load(ts_trend_seasonal):
     transform = FourierTransform(period=7, order=3)
     assert_transformation_equals_loaded_original(transform=transform, ts=ts_trend_seasonal)
+
+
+@pytest.mark.parametrize(
+    "transform",
```
Comment (Mr-Geekman) [inline tests/test_transforms/test_timestamp/test_fourier_transform.py:173]: We should probably also test `FourierTransform(period=7, mods=[1])` but it will fail because during setting `mods=None` there will be no `order` or `mods` remaining.

Comment (Mr-Geekman) [reply tests/test_transforms/test_timestamp/test_fourier_transform.py:174]: I added a new test below.

File: tests/test_transforms/test_timestamp/test_fourier_transform.py [hunk 2]
```diff
@@ -166,3 +168,44 @@ def test_forecast(ts_trend_seasonal):
...
+    [
+        FourierTransform(period=7, mods=[1]),
+        FourierTransform(period=7, mods=[1, 4]),
+    ],
+)
+def test_params_to_tune_with_mods(transform, ts_trend_seasonal):
```
Comment (Mr-Geekman) [inline tests/test_transforms/test_timestamp/test_fourier_transform.py:194]: I'm not really sure does we want make this test instead of the previous one for all models, transforms.

File: tests/test_pipeline/test_pipeline.py [hunk 3]
```diff
@@ -1221,6 +1222,15 @@ def test_predict_return_components(
                 "model.depth": IntUniformDistribution(low=1, high=11, step=1),
                 "model.l2_leaf_reg": LogUniformDistribution(low=0.1, high=200.0),
                 "model.random_strength": LogUniformDistribution(low=1e-05, high=10.0),
+                "transforms.0.day_number_in_week": CategoricalDistribution([False, True]),
```
Comment (alex-hse-repository) [inline tests/test_pipeline/test_pipeline.py:1225]: Why don't you add the same test for ensembles and hierarchical pipeline?

File: tests/test_transforms/test_timestamp/test_fourier_transform.py (hunk 2 above)
Comment (alex-hse-repository) [reply tests/test_transforms/test_timestamp/test_fourier_transform.py:194]: We discussed that it is better to return the empty grid for this case

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: tests/test_pipeline/test_pipeline.py (hunk 3 above)
Comment (Mr-Geekman) [reply tests/test_pipeline/test_pipeline.py:1225]: Because for now ensembles doesn't work with `params_to_tune`, they fail with `NotImplementedError`.

Comment (Mr-Geekman) [reply tests/test_pipeline/test_pipeline.py:1225]: As for `HierarchicalPipeline`, I forgot about it. I'll add the test.

**Review by alex-hse-repository — APPROVED** [review]
